src/lightning/pytorch/strategies/ddp.py

```diff
@@ -433,3 +401,20 @@ def teardown(self) -> None:
             self.model = self._layer_sync.revert(self.model)
 
         super().teardown()
+
+
+class _DDPForwardRedirection(_ForwardRedirection):
+    def on_after_inner_forward(self, wrapper_module: Module, original_module: "pl.LightningModule") -> None:
+        # In manual_optimization, we need to prevent DDP reducer as
+        # it is done manually in `LightningModule.manual_backward`
+
+        # TODO: This should always be DistributedDataParallel, remove _LightningPrecisionModuleWrapperBase
+        assert isinstance(wrapper_module, (DistributedDataParallel, _LightningPrecisionModuleWrapperBase))
```

Comment:
Mypy knows something is fishy here. I'm following up by removing `_LightningPrecisionModuleWrapperBase` to resolve this inconvenience here in a seperate PR.